### PR TITLE
Virtual LLM engine — callback hooks from llama-server into mesh

### LINKED ISSUE #183: Inference pipeline to route to different models
There is a mostly idle piece of code in pipeline.rs which (at time of writing) really only tries to route parts of a request if there are multiple models on one node. 

This would be better done by Shifting it closer to where llama serve is called. Given it is on the mesh, if a request comes in say with image data to a model which doesn’t support it (or tool calling) it could in turn use the mesh to find a complimentary model to explain the image, or flesh out a user request or model response with a different or larger model to enhance the response - if confidence was low (ie making use of the mesh - and any other models available but transparent to the client). This doesn’t have to return back to the proxy as every node is a peer it can discover its own suitable set of partners. Logprob etc coujd be used to decide if it needs to consult another model (or if fast enough, a judge etc) 

Given a mesh can easily have a variety of models available this seems a fun way to increase capability

### PR DESCRIPTION
**Pull request by michaelneale**
This is a big change in some ways: 

1. move our llama.cpp temporary fork to to Mesh-LLM
2. change workflows/tools to use that repo with a specific sha (and instructions for agents to keep it up to date with llama head)
3. enhancements to llama for hooks to trigger when certain inference events happen: 

* if someone tries to use images, audio etc and model doesn't need it, during inference it can reach out on the mesh to find a model who can help 
* after prefil, if the model has high uncertainty - it has the option of asking for help from another model, which will automatically happen:  - Sends the full conversation to a different-architecture peer, gets a "reference answer"  - Injects peer's answer into KV cache as hidden context via temp batch - Purpose: diversity of perspective before generation begins
* if at any time during decode there is an entropy spike above a threshold, it fires a hook which lets the mesh find another model to help out (if model lost), or repetition loop, or sudden loss of confidence

The core idea is that this is interestingly unique to mesh where you can have a diversity of models, so why not let the main model reach out for help if/when it needs to. 

-----
Three callback hooks inside llama-server's generation loop that POST JSON to mesh-llm on localhost. The model being served can ask the mesh for help — caption images, fetch context, verify output — without the caller knowing.

## Hooks

1. **Pre-inference** — before tokenization. mesh-llm sees the full request (messages, images, capabilities). Can inject context synchronously or start async work (image captioning via vision model in mesh).
5. **Post-prefill** — prompt evaluated, first-token logits ready. mesh-llm checks entropy/margin signals. If the model is confused, injects context before the first token.
6. **Pre-response** — generation done, full text available. mesh-llm can verify and inject corrections.

Async consultations return `pending` with an `async_id`. llama-server polls every 16 tokens and injects the result into the live KV cache when ready.

## What changes

- **llama.cpp fork**: `mesh_hook_ctx` struct on slots, signal window, async poll in generation loop, three hook call sites, `--mesh-port` CLI flag
- **mesh-llm**: `POST /mesh/hook` + `GET /mesh/hook/poll/{id}` endpoints, `inference/virtual.rs` decision module, launch.rs passes mesh port

## Status

Design doc only so far: `mesh-llm/docs/VIRTUAL_LLM.md`

Closes #183
Relates to #165

### CODE REVIEW

File: mesh-llm/src/network/openai/transport.rs [hunk 1]
```diff
@@ -1141,6 +1141,48 @@ async fn relay_translated_responses_json<R: AsyncRead + Unpin>(
...
+            rebuilt.push_str(line);
+        }
+        rebuilt.push_str("\r\n");
+    }
+    rebuilt.push_str("\r\n");
+
+    let mut result = rebuilt.into_bytes();
+    result.extend_from_slice(&new_body);
+    *raw = result;
+}
```
[the remark this answers is not in the document]
Comment (michaelneale) [reply mesh-llm/src/network/openai/transport.rs:1184]: wrong.
